### PR TITLE
Improve animation tutorial by adding explaination about the interpolations modes

### PR DESCRIPTION
**Pull request by Ybalrid**
The example only shows and details how to perform `LINEAR` interpolation on vectors.

This PR adds explanations about the other interpolation methods, about how you need to use spherical linear interpolation on quaternion in the `LINEAR`  mode, and how to compute the tangents and the spline interpolation for the `CUBICSPLINE` mode.

### CODE REVIEW

File: gltfTutorial/gltfTutorial_007_Animations.md [hunk 1]
```diff
@@ -116,5 +116,107 @@ In the example above, there are two channels for the animation. Both refer to th
...
+
+Cubic spline intepolation needs more data than just the previous and next keyframe time and values, it also need for each keyframe a couple of tangent vectors that act to smooth out the curve around the keyframe points.
+
+These tangent are stored in the animation channel. For each keyframe described by the animation sampler, the animation channel contains 3 elements : 
+
+ - The input tangent of the keyframe
+ - The keyframe value
+ - The output tangent
+ 
+ The input and output tangents are normaized vectors that will need to be scaled by the duration of the keyframe, we calle that the deltaTime
```
Comment (emackey) [inline gltfTutorial/gltfTutorial_007_Animations.md]: `normaized` => `normalized`, `calle` => `call`

File: gltfTutorial/gltfTutorial_007_Animations.md [hunk 2]
```diff
@@ -116,5 +116,107 @@ In the example above, there are two channels for the animation. Both refer to th
...
+*note: the input tangent of the first keyframe and the output tangent of the last keyframe are totally ignored*
+ 
+To calculate the actual tangents of the keyframe, you need to multiply the direction vectors you got from the channel by `deltaTime` 
+
+```
+    previousTangent = deltaTime * previousOutputTangent
+    nextTangent = deltaTime * nextInputTangent
+```
+ 
+The mathematical funciton is described in the [Apenddix C](https://github.com/KhronosGroup/glTF/tree/master/specification/2.0?ts=4#appendix-c-spline-interpolation) of the glTF 2.0 specification.
```
Comment (emackey) [inline gltfTutorial/gltfTutorial_007_Animations.md]: `Apenddix` => `Appendix`

File: gltfTutorial/gltfTutorial_007_Animations.md [hunk 3]
```diff
@@ -116,5 +116,107 @@ In the example above, there are two channels for the animation. Both refer to th
 <a name="animationChannels-png"></a>Image 7b: Animation channels.
 </p>
 
+## Interpolation
+
+The above example only treated `LINEAR` interpolation. Animations in a glTF asset can use 3 interpolation modes :
```
Comment (emackey) [inline gltfTutorial/gltfTutorial_007_Animations.md]: Suggestion: `treated` => `covers`

**Comment by emackey** [comment]
@javagl Do you have time to take a look at this too?  Here's the [twitter thread](https://twitter.com/Ybalrid/status/1134741355081678848) that prompted this.

**Comment by Ybalrid** [comment]
@emackey I've made the changes you've pointed at

**Comment by Ybalrid** [comment]
I may add a blurb that explicitly states that key-frames in a sampler moving morph targets weights are an array of X weights, X being the number of targets on the morph targets the `target` node as in it's `weights` property.

This is confusing to explain between the "targets", the "target" node and the target blend "weights" for morphing between morph targets...

Maybe the term "blend shape" would have been a better choice in the spec to call a morph target, but we're not here to discuss that anyway...

**Comment by javagl** [comment]
Sorry for the delay here, and thanks for the PR. There are some typos and wording issues. I'll try to have a closer look at this on a technical level ASAP.

File: gltfTutorial/gltfTutorial_007_Animations.md [hunk 4]
```diff
@@ -116,5 +116,107 @@ In the example above, there are two channels for the animation. Both refer to th
...
+
+The above example only covers `LINEAR` interpolation. Animations in a glTF asset can use 3 interpolation modes : 
+
+ - `STEP`
+ - `LINEAR`
+ - `CUBICSPLINE`
+ 
+### Step
+
+The `STEP` interpolation is not really an interpolation mode, it makes objects jumps from keyframe to keyframe *without any sort of interpolation*. When a sampler define a step interpolation, just apply the transformation from the keyframe corresponding to `previousTime`.
```
[the remark this answers is not in the document]
Comment (javagl) [reply gltfTutorial/gltfTutorial_007_Animations.md]: jumps -> jump

**Comment by Ybalrid** [comment]
> Sorry for the delay here, and thanks for the PR. There are some typos and wording issues. I'll try to have a closer look at this on a technical level ASAP.

English is not my primary language, I'll go through what you pointed at and fix the typos. Thanks for taking the time to review! :smile:

**Comment by Ybalrid** [comment]
> I haven't yet done review on a technical level. From quickly skimming over the code snippets, it looks reasonable, but I still have to read the spec about the different interpolation modes more thoroughly. (This is why this section was not covered in detail in the tutorial yet...)
I figured that it was probably still missing because it's a bit tedious to go through all the details. I also feel that the wording of the spec itself may not be the clearest on some points. (e.g: It took some time to me to figure you where *actually* the tangents data were stored in an animation channel when the animation sampler uses `CUBICSPLINE`)

**Comment by Ybalrid** [comment]
@javagl Thank you so much for proofreading my little mess full of typos and my sometimes too frenchy wording 🙂 

I just implemented your suggested changes

**Comment by javagl** [comment]
It still says `dotProcut` in the code, instead of `dotProduct`. Apart from that, I think that it can be merged, unless there are further comments. (I'd like to review the spline interpolation part in more detail, but cannot give a timeline here. In doubt, I might extend it with some images or examples later).

**Comment by Ybalrid** [comment]
Okay, I'm fixing that, I've cleaned up a bit so that the PR doesn't make a mess in the commit history

**Pull request merged**